internal/rest/v1/project.go

```diff
@@ -249,3 +249,39 @@ func (h *Handlers) newGetProjectVersionsResponse(
 	return response
 }
 
+type deleteProjectVersionRequest struct {
+	ProjectID uint `param:"project_id" validate:"required"`
+	VersionID uint `param:"version_id" validate:"required"`
+}
+
+func (h *Handlers) deleteProjectVersion(ctx echo.Context, l *logrus.Entry) error {
+	req := new(deleteProjectVersionRequest)
+	if err := ctx.Bind(req); err != nil {
+		l.WithError(err).Error("Error binding request")
+
+		return echo.ErrBadRequest
+	}
+
+	l = l.WithFields(logrus.Fields{
+		"project": req.ProjectID,
+		"version": req.VersionID,
+	})
+
+	err := h.ProjectService.DeleteProjectVersionByIDAndProjectID(
+		ctx.Request().Context(),
+		req.VersionID,
+		req.ProjectID,
+	)
+	if err != nil {
+		if (err.Error() == "not found") {
+			return echo.ErrNotFound
+		}
```

Comment:
Instead of checking the error message, check the error with `errors.Is` against `ErrNotFound` that you defined in `internal/project/repository.go`